cmd/harmony/main.go

```diff
@@ -157,37 +194,85 @@ func raiseFdLimits() error {
 	return nil
 }
 
-func getHarmonyConfig(cmd *cobra.Command) (harmonyConfig, error) {
+func promptForConfigUpgrade() bool {
+	if !terminal.IsTerminal(syscall.Stdin) {
+		return false
+	}
+	for {
```

Comment:
Would you please add a timeout mechanism if possible?

Reply:
do you want for this prompt to disappear in some time if user does not take any action? Frankly speaking, I never saw this kind of behavior in cli applications before. or do you want me to limit number of attempts that use can mistype their choice?

IMHO if we know for sure that user is running this in the terminal, this current behavior is what one might expect for interactive cli application. but I am opened to your suggestions, if you elaborate, please.

Reply:
There is a legacy way of starting harmony node with shell script `node.sh`: https://github.com/harmony-one/harmony/blob/main/scripts/node.sh

The suggested way of running harmony at that time was to run the shell script within tmux session. Though starting with `node.sh` is not recommended since a long time ago, we still spot some users keeps running nodes with deprecated `node.sh`. Thus the prompt without a timeout will potentially cause trouble in this case. 

Though it's uncommon, I would still suggest you to add a timeout mechanism if possible. Thanks!